torch/utils/data/dataloader.py

```diff
@@ -252,23 +258,25 @@ def __init__(self, loader):
             self.worker_init_fn = loader.worker_init_fn
             self.index_queues = [multiprocessing.Queue() for _ in range(self.num_workers)]
             self.worker_queue_idx = 0
-            self.worker_result_queue = multiprocessing.SimpleQueue()
+            self.worker_result_queue = multiprocessing.Queue()
```

Comment:
Any reason to prefer a `Queue` here?

Reply:
So that in workers `data_queue.put` is non-blocking and they can call `data_queue.cancel_join_thread`

Reply:
Yeah this makes me a bit nervous. The reason why we used a `SimpleQueue` here is precisely because `fork + threading = trouble`, and you depend on some pretty subtle behavior here (that fork preserves the threads!), which is undocumented and might not be the case I think. On the other hand it's possible that `multiprocessing` is smart enough to do the right thing when you pass the queues around...

Reply:
:) I think `multiprocessing` should be smart enough. After all, this is called `multiprocessing.Queue`, i.e., a `queues.Queue` particularly designed to pass data around processes. If this isn't supported, then it wouldn't be any useful.

Looking at `cpython` source, I think this is supported indeed: https://github.com/python/cpython/blob/3.6/Lib/multiprocessing/queues.py#L67-L90